### PR TITLE
Minor improvements to Utils package

### PR DESCRIPTION
**Pull request by andrew-fleming**
#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Added entry to CHANGELOG.md 
- [ ] Tried the feature on a public network 

This PR proposes to add fuzz tests for the math module (which only includes the `average` fn ATM)

### CODE REVIEW

File: packages/utils/src/structs/checkpoint.cairo [hunk 1]
```diff
@@ -99,12 +99,12 @@ pub impl TraceImpl of TraceTrait {
         }
     }
 
-    /// Returns the number of checkpoints.
+    /// Returns the total number of checkpoints.
     fn length(self: StoragePath<Trace>) -> u64 {
         self.checkpoints.len()
     }
 
-    /// Returns the checkpoint at given position.
```
Comment (ericnordelo) [inline packages/utils/src/structs/checkpoint.cairo:107]: I think position is easier to read here, maybe we can update the param name?

Comment (andrew-fleming) [reply packages/utils/src/structs/checkpoint.cairo:107]: Mmm I honestly like `pos` as the param name—unless you prefer a new name? I'll switch the comment back. Looking at it now, I agree position is easier to read

File: packages/utils/src/tests/test_math.cairo [hunk 2]
```diff
@@ -0,0 +1,75 @@
+use core::integer::{u512, u512_safe_div_rem_by_u256};
+use core::num::traits::OverflowingAdd;
+use crate::math::average;
+
+#[test]
+fn test_average_u8(a: u8, b: u8) {
```
Comment (ericnordelo) [inline packages/utils/src/tests/test_math.cairo:6]: I noticed that fuzzing actually takes a significant amount of time when running tests, not only in compilation. Did you noticed the same thing?

Comment (andrew-fleming) [reply packages/utils/src/tests/test_math.cairo:6]: I did...it's only a few tests though so relative to the whole repo, IMO it's not a huge deal. If we want to improve performance, we can configure it to a smaller amount either in the tests themselves or globally when running the test cmd i.e. `snforge test ... --fuzzer-runs 25`

File: packages/utils/src/cryptography/snip12.cairo [hunk 3]
```diff
@@ -47,6 +51,14 @@ pub trait SNIP12Metadata {
     fn version() -> felt252;
 }
 
+/// Implementation of OffchainMessageHash that calculates the Poseidon hash of the message.
+///
+/// The hash state hashes the following in order:
+///
+/// - 'StarkNet Message' short string.
+/// - Starknet domain.
```
Comment (ericnordelo) [inline packages/utils/src/cryptography/snip12.cairo:59]:
```suggestion
/// - Starknet domain struct hash.
```

**Review by ericnordelo — APPROVED** [review]
Left a small comment, looks good to me otherwise!

**Review by immrsd — APPROVED** [review]
LGTM

**Pull request merged**